src/aws_sync.js

```diff
@@ -20,21 +20,34 @@ const downloadCourses = async (coursesJson, coursesDir) => {
   if (!coursesDir) {
     throw new Error("Invalid courses directory")
   }
+  if (!process.env["AWS_BUCKET_NAME"]) {
+    throw new Error("AWS_BUCKET_NAME not set")
+  }
+  /**
+   * We explicitly configure credentials here if we see them set because
+   * the aws-sdk will not automatically configure itself from values set in
+   * .env files
+   */
   if (
-    !env["AWS_REGION"] ||
-    !env["AWS_BUCKET_NAME"] ||
-    !env["AWS_ACCESS_KEY"] ||
-    !env["AWS_SECRET_ACCESS_KEY"]
+    process.env["AWS_REGION"] &&
+    process.env["AWS_ACCESS_KEY"] &&
+    process.env["AWS_SECRET_ACCESS_KEY"]
   ) {
-    throw new Error("Invalid AWS connection info")
+    const config = {}
+    if (process.env["AWS_REGION"]) {
+      config["region"] = process.env["AWS_REGION"]
+    }
+    if (process.env["AWS_ACCESS_KEY"] && process.env["AWS_SECRET_ACCESS_KEY"]) {
```

Comment:
it looks like this `if` block (and the one above on line 37) is redundant because of the one on line 31

Reply:
Ah yea, I flipped the logic on it to require all of them to be set and forgot about that...